core/src/consensus/tendermint/mod.rs

```diff
@@ -203,8 +215,8 @@ impl TendermintInner {
     }
 
     /// Find the designated for the given view.
-    fn view_proposer(&self, bh: &H256, height: Height, view: View) -> Address {
-        let proposer_nonce = height + view;
+    fn view_proposer(&self, bh: &H256, _height: Height, view: View) -> Address {
+        let proposer_nonce = self.prev_block_proposer_idx() + 1 + view;
```

Comment:
The `view_proposer` receives the height, view, and prev block's hash, and returns the propser's address.
We should use the give prev block's hash instead of the (current height -1)th block's hash.

Please remind that this equation `received height == tendermint.height` is not always true.

Reply:
Fixed.